### DISCUSSION TITLE
[Question] Can I combine use DynamicVisibility and BeforeQueryStatus ?

### DISCUSSION THREAD

**Discussion by phuocle** [VsixCommunity/Community.VisualStudio.Toolkit#196] [Q&A]
I can use the `<CommandFlag>DynamicVisibility</CommandFlag>` that show/hide the popup window code menu with the extension `.cs` and `.vb` correctly. (Thanks @madskristensen for [the sample](https://github.com/VsixCommunity/Samples/tree/master/SingleFileGenerator))

But on the popup, I want to use: `BeforeQueryStatus` because I want to check the current position in the `class` and class should inherit by an `interface`

My old code working fine with `BeforeQueryStatus`  (and sure I don't use `DynamicVisibility`)
It used `ProvideAutoLoad` that I don't want to use, it make my extension take time to load.

Any help or sample with the combine `DynamicVisibility` and `BeforeQueryStatus` wil be great. Tried a lot of cases but not success.

Thanks.

PS: On the way convert all existing code to `Community.VisualStudio.Toolkit`, great lib.

**Comment by madskristensen** [comment]
You can only use BeforeQueryStatus once your package has been loaded. So if you need to use it, you should set the `ProvideAutoload` attribute on your package class
